src/DockWidget.cpp

```diff
@@ -515,23 +516,36 @@ void CDockWidget::flagAsUnassigned()
 //============================================================================
 bool CDockWidget::event(QEvent *e)
 {
-	if (e->type() == QEvent::WindowTitleChange)
-	{
-		const auto title = windowTitle();
-		if (d->TabWidget)
-		{
-			d->TabWidget->setText(title);
-		}
-		if (d->ToggleViewAction)
-		{
-			d->ToggleViewAction->setText(title);
-		}
-		if (d->DockArea)
-		{
-			d->DockArea->markTitleBarMenuOutdated();//update tabs menu
-		}
-		emit titleChanged(title);
-	}
+    switch (e->type())
+    {
+#ifndef QT_NO_ACTION

```

Comment:
Whats the reason for wrapping the signal into the QT_NO_ACTION macro?

Reply:
No reason, you are right. I just borrowed code from QDockWidget and the QT_NO_ACTION was set but we don't need it here: https://code.qt.io/cgit/qt/qtbase.git/tree/src/widgets/widgets/qdockwidget.cpp?h=dev#n1517